app/models/manageiq/providers/storage_manager.rb

```diff
@@ -24,5 +25,20 @@ class << model_name
       define_method(:route_key) { "ems_storages" }
       define_method(:singular_route_key) { "ems_storage" }
     end
+
+    def model_feature_for_action(action)
+      case action
+      when :edit
+        # Only storage managers that can be added directly from the storage section can be edit.
+        :ems_storage_new
+      end
+    end
+
+    # Allow only adding supported types. Non-supported types for adding will not be visible in the Type field
+    def self.supported_types_and_descriptions_hash
+      supported_subclasses.select(&:supports_ems_storage_new?).each_with_object({}) do |klass, hash|
+        hash[klass.ems_type] = klass.description
+      end
+    end
```

Comment:
Note to self, I'd like to move the `belongs_to :parent_manager` to a mixin which would drop the need for this in the NetworkManager and here